pyteal/ast/txn.py

```diff
@@ -161,6 +161,15 @@ def type_of(self):
 class TxnaExpr(LeafExpr):
     """An expression that accesses a transaction array field from the current transaction."""
 
+    @staticmethod
+    def __validate_index_or_throw(index: Union[int, Expr]):
+        if not isinstance(index, (int, Expr)):
+            raise TealInputError(
+                f"Invalid index type:  Expected int or Expr, but received {index}."
+            )
+        if isinstance(index, Expr):
+            require_type(index, TealType.uint64)
+
```

Comment:
nit: I believe this function has a same implementation in `gtxn.py` (which is written by you), right? I believe this can be reused somehow?

Reply:
@ahangsu I agree the method can be reworked to abstract over the field (`index` vs `txn_index`).  Though, I don't know where it's appropriate for a common definition to live.  Do you have a suggestion?

After considering a few options, I don't feel strongly that the duplication ought to be addressed now.  But, I'm on-board to make a change if you have a low effort suggestion that maintains repo organizational norms.

The only _plausible_ suggestion I came away with is:  
* Define static methods in `LeafExpr` to support validation for select fields (`index` and `txn_index` to start).
* It's unclear to me if adding validation methods to `LeafExpr` fits with the existing repo organization.

I also considered:
* In my mind, a common definition ought to live with the definition of the type (`Union[int, Expr]`).  Since a custom type doesn't exist, there isn't an associated place to define the validation.
* Alternatively, I can imagine defining a more generic validation object with an arbitrary `isinstance` check for all members of a `Union` data type.
  * I don't know enough about Python to know how to write the check.  It feels to me a bit over-the-top for the problem at-hand.

Reply:
hmmm... I don't think defining static method in `LeafExpr` is a good option, since some other stuffs might inherit from `LeafExpr`, which "contaminates" the basic definition.

An idea I am thinking of:, extending `validate_txn_index_or_throw` to allow optional default argument, i.e., of form
```python
def validate_txn_index_or_throw(txnIndex: Union[int, Expr], fieldName: str = "txn_index"):
```
and probably rename the function to a most general `validate_index_or_throw`. This gives a low effort modification, and saves some lines of somewhat duplicated definition.

As per the "more generic validation object", I agree the idea is appealing, but it might take too much effort to get into this. For the spirit of this PR, I don't feel strong about that...

Reply:
Yea I am convinced that it might causing more obfuscation than simplicity here, providing the static method here is fine for me.